tests/ui/panic_unimplemented.rs

```diff
@@ -65,4 +77,6 @@ fn main() {
     ok_nomsg();
     ok_escaped();
     unimplemented();
+    unreachable();
```

Comment:
Since you separated out `panic!`, can you also separate these test cases?

Reply:
I would have left panic in but panic and panic with sensible message clash.

Reply:
We want to split up the tests as much as possible, see #2038. Since the unreachable, panic, unimplemented and todo tests are pretty small, it may make sense to group them in a `panicking_macros.rs` test file.